src/module/entityTypes/helper.js

```diff
@@ -7,96 +7,289 @@
 
 // Dependencies
 
-
 const { v4: uuidv4 } = require('uuid')
 
 // const entitiesHelper = require(MODULES_BASE_PATH + "/entities/helper")
-const _ = require("lodash")
+const _ = require('lodash')
 // const programUsersQueries = require(DB_QUERY_BASE_PATH + "/programUsers")
 
 /**
-    * UserProjectsHelper
-    * @class
-*/
+ * UserProjectsHelper
+ * @class
+ */
 
 module.exports = class UserProjectsHelper {
+	static bulkCreate(entityTypesCSVData, userDetails) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				const entityTypesUploadedData = await Promise.all(
+					entityTypesCSVData.map(async (entityType) => {
+						try {
+							entityType = gen.utils.valueParser(entityType)
+							entityType.regsitryDetails = {}
+							let removedKeys = []
+							Object.keys(entityType).forEach(function (eachKey) {
+								if (eachKey.startsWith('registry-')) {
+									let newKey = eachKey.replace('registry-', '')
+									entityType.regsitryDetails[newKey] = entityType[eachKey]
+									removedKeys.push(entityType[eachKey])
+								}
+							})
+							if (entityType.profileFields) {
+								entityType.profileFields = entityType.profileFields.split(',') || []
+							}
+
+							if (
+								entityType.immediateChildrenEntityType != '' &&
+								entityType.immediateChildrenEntityType != undefined
+							) {
+								let entityTypeImmediateChildren = entityType.immediateChildrenEntityType.split(',')
+								entityTypeImmediateChildren = _.uniq(entityTypeImmediateChildren)
+
+								entityType.immediateChildrenEntityType = new Array()
+								entityTypeImmediateChildren.forEach((immediateChildren) => {
+									entityType.immediateChildrenEntityType.push(immediateChildren)
+								})
+							}
+
+							if (entityType.isObservable) {
+								entityType.isObservable = gen.utils.convertStringToBoolean(entityType.isObservable)
+							}
+							if (entityType.toBeMappedToParentEntities) {
+								entityType.toBeMappedToParentEntities = gen.utils.convertStringToBoolean(
+									entityType.toBeMappedToParentEntities
+								)
+							}
+
+							if (removedKeys && removedKeys.length > 0) {
+								for (var key in entityType) {
+									for (var removedKey in removedKeys) {
+										if (entityType.hasOwnProperty(removedKey)) {
+											delete entityType[removedKey]
+										}
+									}
+								}
+							}
+							let uniqId
+							if (userDetails && userDetails.id) {
```

Comment:
@MallanagoudaB you can have this check in one line 
const uniqId = userDetails && userDetails.id? userDetails && userDetails.id : common.SYSTEM

Reply:
And its not uniqId the variable name is userId